File: BAEKJOON/java/5789_한다_안한다.java
import java.io.*;
import java.util.StringTokenizer;

public class Main {
    public static String s;

    public static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    public static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    public static void main(String[] args) throws IOException {
        int tc = Integer.parseInt(br.readLine());

        for (int i = 0; i < tc; i++) {
            input();
            solution();
        }

        bw.flush();
        bw.close();
    }

    private static void input() throws IOException {
        s = br.readLine();
    }

    private static void solution() throws IOException {
       int mid = (s.length() / 2) - 1;

       if (s.charAt(mid) == s.charAt(mid + 1)) {
           bw.write("Do-it\n");
           return;
       }

       bw.write("Do-it-Not\n");
    }
}

File: BAEKJOON/cpp/5789_한다_안한다.cpp
#include <iostream>
#include <string>
using namespace std;

string s;

void input() {
	cin >> s;
}

void solution() {
	int mid = (s.length() / 2) - 1;

	if (s[mid] == s[mid + 1]) {
		cout << "Do-it\n";
		return;
	}

	cout << "Do-it-Not\n";
}

int main() {
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	int tc;
	cin >> tc;

	for (int i = 0; i < tc; i++) {
		input();
		solution();
	}

	return 0;
}